src/gluonnlp/utils/parallel.py

```diff
@@ -130,7 +130,7 @@ def _worker(in_queue, out_queue, parallel):
     def put(self, x):
         """Assign input `x` to an available worker and invoke
         `parallizable.forward_backward` with x. """
-        if self._num_serial > 0:
+        if self._num_serial > 0 or len(self._threads) == 0:
```

Comment:
Why `len(self._threads) == 0`?

Reply:
if `self._num_workers  == 0` then `len(self._threads)  == 0` and we will execute forward_backward using the main thread